src/javascript/JContent/ContentNavigation/NavigationHeader/SwitchersLayout/SiteSwitcher.jsx

```diff
@@ -43,48 +40,49 @@ class SiteSwitcher extends React.Component {
             ${PredefinedFragments.nodeCacheRequiredFields.gql}
         `;
 
-        this.onSelectSite = this.onSelectSite.bind(this);
-    }
-
-    getSites(data) {
-        let siteNodes = [];
-        if (data && data.jcr.result !== null) {
-            for (let i in data.jcr.result.siteNodes) {
-                if (data.jcr.result.siteNodes[i].hasPermission) {
-                    siteNodes.push(data.jcr.result.siteNodes[i]);
-                }
+const getSites = data => {
+    let siteNodes = [];
+    if (data && data.jcr.result !== null) {
+        for (let i in data.jcr.result.siteNodes) {
+            if (data.jcr.result.siteNodes[i].hasPermission) {
+                siteNodes.push(data.jcr.result.siteNodes[i]);
             }
         }
-
-        return _.sortBy(siteNodes, ['displayName']);
     }
 
-    getTargetSiteLanguageForSwitch(siteNode, currentLang) {
-        let newLang = null;
-        let siteLanguages = siteNode.site.languages;
-        for (let i in siteLanguages) {
-            if (Object.prototype.hasOwnProperty.call(siteLanguages, i)) {
-                let lang = siteLanguages[i];
-                if (lang.activeInEdit && lang.language === currentLang) {
-                    newLang = currentLang;
-                    break;
-                }
+    return _.sortBy(siteNodes, ['displayName']);
+};
+
+const getTargetSiteLanguageForSwitch = (siteNode, currentLang) => {
+    let newLang = null;
+    let siteLanguages = siteNode.site.languages;
+    for (let i in siteLanguages) {
+        if (Object.prototype.hasOwnProperty.call(siteLanguages, i)) {
+            let lang = siteLanguages[i];
+            if (lang.activeInEdit && lang.language === currentLang) {
+                newLang = currentLang;
+                break;
             }
         }
-
-        return newLang ? newLang : siteNode.site.defaultLanguage;
     }
 
-    onSelectSite(siteNode, currentLang) {
-        let newLang = this.getTargetSiteLanguageForSwitch(siteNode, currentLang);
-        this.props.selectSite(siteNode, newLang);
-    }
+    return newLang ? newLang : siteNode.site.defaultLanguage;
+};
+
+const SiteSwitcher = ({selectorObject, onSelectAction, notificationContext}) => {
+    const {t} = useTranslation();
+    const selector = useMemo(() => createStructuredSelector(selectorObject), [selectorObject]);
+    const dispatch = useDispatch();
+    const {siteKey, currentLang} = useSelector(state => selector(state));
```

Comment:
Same remark as in ContentNavigation - directly passing the selector function and avoiding reselect will be more clear

Reply:
replied above